frontend/src/leaderboard.js

```diff
@@ -1,36 +1,87 @@
-const ACCOUNTS_URL = 'http://localhost:3000/accounts';
 
-document.getElementById('leaderboard-button').addEventListener('click', () => {
-  const leaderboard = document.getElementById('leaderboard-window');
-  while (leaderboard.firstChild) {
-    leaderboard.removeChild(leaderboard.firstChild);
+class Leaderboard {
+  constructor() {
+    this.leaderboardDiv = document.getElementById('users-container');
+    this.leaderboardWindow = document.getElementById('leaderboard-window');
+    this.leaderboardToggleButton = document.getElementById('leaderboard-toggle-button');
+
+    this.displayAccount = this.displayAccount.bind(this);
+    this.displayAccounts = this.displayAccounts.bind(this);
+    this.displayAccountBestRun = this.displayAccountBestRun.bind(this);
+    this.createAccountDiv = this.createAccountDiv.bind(this);
+    this.getBestRun = this.getBestRun.bind(this);
+    this.createBestRunElements = this.createBestRunElements.bind(this);
+    this.fetchUsers = this.fetchUsers.bind(this);
+    this.handleLeaderboardToggle = this.handleLeaderboardToggle.bind(this);
```

Comment:
A bit confused by this. From what I have read from [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) it looks like it's used to set the this context.

Since these are functions that you want to call that exist within the class can you not simply call them by name?

Maybe I just don't understand how this works though.

Reply:
@justinamaple 
Yeah I agree it's kind of ugly. I made the leaderboard into a class and when I would use `this` it would refer to a DOM element or the global obj. 
There is definitely a way to remove all of the `bind(this)` statements. I can wrap everything in an arrow function (I guess arrow functions provide the proper `this` context) however I don't think I can remove the `this.` in front of the class methods without them erroring out. Feels weird and redundant to type but I think its necessary? 
I'm new to this too haha.
Would y'all prefer arrow functions? @kel29  If so, I'll tweak this
cheers.